funcx_endpoint/funcx_endpoint/executors/high_throughput/interchange.py

```diff
@@ -587,6 +597,26 @@ def _command_server(self, kill_event):
                 logger.debug("[COMMAND] is alive")
                 continue
 
+    def enqueue_task_cancel(self, task_id):
+        """ Cancel a task on the interchange """
+        logger.debug(f"Received task_cancel request for Task:{task_id}")
+
+        found_task = False
+        for manager in self._ready_manager_queue:
```

Comment:
does this block have a race condition where it detects that a task is pending, but by the time it processes that pending message, the task is now running?

Reply:
I believe setting the trap ahead of the search guarantees that we won't miss the task. I've added some notes to explain this better.